config/main.py

```diff
@@ -6854,6 +6856,289 @@ def del_interface_storm(ctx,port_name,storm_type, namespace):
+#
+@authentication.command('order')
+@click.pass_context
+@click.argument('interface_name', metavar='<interface_name>', required=True)
+@click.argument('order_list', metavar='<order_list>', required=True, type=click.Choice(['dot1x,mab', 'mab,dot1x', 'dot1x', 'mab']))
+def add_order(ctx, order_list, interface_name):
+    """Add order configutation"""
+
+    config_db = ValidatedConfigDBConnector(ConfigDBConnector())
+    config_db.connect()
+
+    order_list.split()
+    try:
+        config_db.mod_entry('PAC_PORT_CONFIG_TABLE', (interface_name), {'method_list@': order_list })
+
+    except ValueError as e:
+        ctx.fail("Invalid ConfigDB. Error: {}".format(e))
+
+
+#
+# 'authentication priority' command ('config interface authentication priority <interface> <priority_list>')
+#
+@authentication.command('priority')
+@click.pass_context
+@click.argument('interface_name', metavar='<interface_name>', required=True)
+@click.argument('priority_list', metavar='<priority_list>', required=True, type=click.Choice(['dot1x,mab', 'mab,dot1x', 'dot1x', 'mab']))
+def add_priority(ctx, priority_list, interface_name):
+    """Add priority configutation"""
+
+    config_db = ValidatedConfigDBConnector(ConfigDBConnector())
+    config_db.connect()
+
+    priority_list.split()
+    try:
+        config_db.mod_entry('PAC_PORT_CONFIG_TABLE', (interface_name), {'priority_list@': priority_list })
+
+    except ValueError as e:
+        ctx.fail("Invalid ConfigDB. Error: {}".format(e))
+
+#
+# 'mab' group ('config interface mab ...')
+#
+@interface.group('mab')
+@click.pass_context
+def mab(ctx):
+    """Mac auth bypass authentication-related configuration tasks"""
+    pass
+
+
+
+#
+# 'mab' command ('config interface mab <interface> <enable|disable> [eap-md5|pap|chap]')
+#
+@interface.command('mab')
+@click.pass_context
+@click.argument('interface_name', metavar='<interface_name>', required=True)
+@click.argument('status', metavar='<status>', required=True, type=click.Choice(['enable', 'disable']))
+@click.option('-a', '--auth-type', required=False, type=click.Choice(['eap-md5', 'pap', 'chap']))
+def add_mab(ctx, status, auth_type, interface_name):
+    """Add mab configutation"""
+
+    config_db = ValidatedConfigDBConnector(ConfigDBConnector())
+    config_db.connect()
+
+    print(status)
+    print(auth_type)
+    print(interface_name)
+    try:
+        config_db.mod_entry('MAB_PORT_CONFIG_TABLE', (interface_name), {'mab_enable': True if status == 'enable' else False})
+        if auth_type == 'pap' or auth_type == 'chap':
+            config_db.mod_entry('MAB_PORT_CONFIG_TABLE', (interface_name), {'mab_auth_type': auth_type})
+        else:
+            config_db.mod_entry('MAB_PORT_CONFIG_TABLE', (interface_name), {'mab_auth_type': 'eap-md5'})
+
+    except ValueError as e:
+        ctx.fail("Invalid ConfigDB. Error: {}".format(e))
+
+
+def pac_is_port_any_vlan_member(config_db, port):
```

Comment:
While a port is dot1X enabled, it should not be allowed to become a router-port. In other words, it should not be removed from all the vlans right. Is that taken care as part of seperate PR?